zipline/_protocol.pyx

```diff
@@ -508,6 +508,9 @@ cdef class BarData:
             # asset isn't alive
             return False
 
+        if asset.auto_close_date and dt >= asset.auto_close_date:
```

Comment:
Not sure; if `is_alive_for_session` is strictly meant to check if we're between the asset's start and end date then probably not, considering a future's auto close date is a few days before the end date (at which point it could still be called "alive"). If we are being more lenient though, then yes I'd say this logic could be moved to `is_alive_for_session`.